classification/classification_network.py

```diff
@@ -166,189 +115,356 @@ def createSensorRegion(network, sensorType, encoders, dataSource, numCats):
-    "py.CLAClassifierRegion"; possible options can be found in /nupic/regions/.
-   
-  @return (Region) Classifier region of the network.
+  @param network: (Network) regions to be linked are nodes in this network.
+  @param sensorRegionName: (str) name of the sensor region
+  @param previousRegion: (PyRegion) parent node in the network
+  @param currentRegion: (PyRegion) current node in the network
+  """
+  network.link(previousRegion, currentRegion, "UniformLink", "")
+  network.link(sensorRegionName, currentRegion, "UniformLink", "",
+               srcOutput="resetOut", destInput="resetIn")
+
+
 
+def _validateRegionWidths(previousRegionWidth, currentRegionWidth):
+  """
+  Make sure previous and current region have compatible input and output width
+  
+  @param previousRegionWidth: (int) width of the previous region in the network
+  @param currentRegionWidth: (int) width of the current region
   """
-  # Classifier region may be non-standard, so add custom region class to the 
-  # network
-  if classifierType.split(".")[1] not in PY_REGIONS:
-    # Add new region class to the network
-    network.registerRegion(SequenceClassifierRegion)
-    PY_REGIONS.append(classifierType.split(".")[1])
 
-  # Create the classifier region.
-  classifierRegion = network.addRegion(
-    "classifier", classifierType, json.dumps(classifierParams))
+  if previousRegionWidth != currentRegionWidth:
+    raise ValueError("Region widths do not fit. Output width = {}, "
+                     "input width = {}.".format(previousRegionWidth,
+                                                currentRegionWidth))
 
-  # Disable learning for now (will be enabled in a later training phase)
-  classifierRegion.setParameter("learningMode", False)
 
-  # Okay to always leave inference mode on; only there for some corner cases.
-  classifierRegion.setParameter("inferenceMode", True)
 
-  return classifierRegion
+def createNetwork(dataSource,
+                  networkConfiguration):
+  """
+  Create and initialize the network instance with regions for the sensor, SP, 
+  TM, and classifier. Before running, be sure to init w/ network.initialize().
 
+  @param dataSource: (RecordStream) Sensor region reads data from here.
+  @param networkConfiguration: (dict) the configuration of this network. 
+  @return network: (Network) Sample network. E.g. Sensor -> SP -> TM -> Classif.
+  """
+  network = Network()
 
+  # Create sensor regions (always enabled)
+  sensorParams = networkConfiguration[SENSOR_REGION_NAME]
+  sensorRegion = _createSensorRegion(network,
+                                     sensorParams,
+                                     dataSource)
+
+  # Keep track of the previous region name and width to validate and link the 
+  # input/output width of two consecutive regions.
+  previousRegion = SENSOR_REGION_NAME
+  previousRegionWidth = sensorRegion.encoder.getWidth()
+
+  # Create SP region, if enabled.
```

Comment:
These if blocks should be generalized into one `createRegion(regionParams)` method.